core/src/com/fourinachamber/fortyfive/map/statusbar/Backpack.kt

```diff
@@ -1,32 +1,427 @@
 package com.fourinachamber.fortyfive.map.statusbar
 
-import com.fourinachamber.fortyfive.screen.general.CustomFlexBox
-import com.fourinachamber.fortyfive.screen.general.OnjScreen
-import com.fourinachamber.fortyfive.screen.general.InOutAnimationActor
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.math.Interpolation
+import com.badlogic.gdx.scenes.scene2d.InputEvent
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
+import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable
+import com.fourinachamber.fortyfive.game.SaveState
+import com.fourinachamber.fortyfive.game.card.Card
+import com.fourinachamber.fortyfive.game.card.CardPrototype
+import com.fourinachamber.fortyfive.screen.general.*
+import com.fourinachamber.fortyfive.screen.general.customActor.CustomInputField
+import com.fourinachamber.fortyfive.screen.general.customActor.CustomMoveByAction
+import com.fourinachamber.fortyfive.screen.general.customActor.CustomWarningParent
+import com.fourinachamber.fortyfive.utils.FortyFiveLogger
+import com.fourinachamber.fortyfive.utils.TemplateString
 import com.fourinachamber.fortyfive.utils.Timeline
+import ktx.actors.onClick
+import onj.parser.OnjParser
+import onj.parser.OnjSchemaParser
+import onj.schema.OnjSchema
+import onj.value.OnjArray
+import onj.value.OnjObject
+
+class Backpack(
+    screen: OnjScreen,
+    cardsFile: String,
+    backpackFile: String,
+    private val deckNameWidgetName: String,
+    private val deckSelectionParentWidgetName: String,
+    private val deckCardsWidgetName: String,
+    private val backPackCardsWidgetName: String,
+    private val backpackEditIndicationWidgetName: String,
+    private val sortWidgetName: String,
+    private val sortReverseWidgetName: String,
+) :
+    CustomFlexBox(screen),
+    InOutAnimationActor {
+
+    private val cardPrototypes: List<CardPrototype>
+    private val _allCards: MutableList<Card>
+    private val minNameSize: Int
+    private val maxNameSize: Int
+
+    private lateinit var deckNameWidget: CustomInputField
+    private lateinit var deckCardsWidget: CustomScrollableFlexBox
+    private lateinit var backpackCardsWidget: CustomScrollableFlexBox
+    private lateinit var deckSelectionParent: CustomFlexBox
+    private lateinit var backpackEditIndication: CustomFlexBox
+
+    private var sortingSystem: BackpackSorting = BackpackSorting.Damage()
+
+    private val quickAddRemoveListener = object : ClickListener() {
+        override fun clicked(event: InputEvent?, x: Float, y: Float) {
+            super.clicked(event, x, y)
+            val targetName = (event!!.target as CustomFlexBox).name.split(NAME_SEPARATOR_STRING)
+            if ((tapCount and 1) == 0) {
```

Comment:
please just use modulo

Reply:
the jit compiler will just optimize it anyway, no need to care about a few clock cycles